eva_submission/eload_submission.py

```diff
@@ -191,27 +191,29 @@ def detect_submitted_vcf(self):
         vcf_files = uncompressed_vcf + compressed_vcf
         if len(vcf_files) < 1:
             raise FileNotFoundError('Could not locate vcf file in in %s', vcf_dir)
-        self.eload_cfg.set('submission','vcf_files', value=vcf_files)
+        self.eload_cfg.set('submission', 'vcf_files', value=vcf_files)
 
     def detect_metadata_attributes(self):
         eva_metadata = EvaXlsxReader(self.eload_cfg.query('submission', 'metadata_spreadsheet'))
         reference_set = set()
+        analysis_reference = {}
         for analysis in eva_metadata.analysis:
-            reference_txt = analysis.get('Reference')
-            if reference_txt:
-                reference_set.update(resolve_accession_from_text(reference_txt))
+            assembly_accession = analysis.get('Reference')
+            if assembly_accession:
+                reference_set.update(assembly_accession)
+                analysis_reference[analysis.get('Analysis Alias')] = {'assembly_accession': assembly_accession,
+                                                                      'vcf_files': []}
             else:
-                self.warning('Reference is missing for Analysis %s', analysis.get('Analysis Alias'))
+                self.error(f"Reference is missing for Analysis {analysis.get('Analysis Alias')}")
 
-        if len(reference_set) > 1:
-            self.warning('Multiple assemblies in project: %s', ', '.join(reference_set))
-            self.warning('Will look for the most recent assembly.')
-            reference_set = {sorted(reference_set)[-1]}
+        for file in eva_metadata.files:
+            if file.get("File Type") == 'vcf':
+                file_full = os.path.join(self.eload_dir, directory_structure['vcf'], file.get("File Name"))
+                analysis_alias = file.get("Analysis Alias")
+                analysis_reference[analysis_alias]['vcf_files'].append(file_full)
+                self.eload_cfg.set('submission', 'analyses', analysis_alias, 'vcf_files', value=file_full)
```

Comment:
I think you don't need this `set`, since in the end you just set the entire submission -> analyses part of the config.